enigma/src/main/java/org/quiltmc/enigma/api/analysis/index/mapping/MappingsIndex.java

```diff
@@ -0,0 +1,108 @@
+package org.quiltmc.enigma.api.analysis.index.mapping;
+
+import org.quiltmc.enigma.api.ProgressListener;
+import org.quiltmc.enigma.api.translation.mapping.EntryMapping;
+import org.quiltmc.enigma.api.translation.mapping.tree.EntryTree;
+import org.quiltmc.enigma.api.translation.mapping.tree.EntryTreeNode;
+import org.quiltmc.enigma.api.translation.representation.entry.ClassEntry;
+import org.quiltmc.enigma.api.translation.representation.entry.Entry;
+import org.quiltmc.enigma.api.translation.representation.entry.FieldEntry;
+import org.quiltmc.enigma.api.translation.representation.entry.LocalVariableEntry;
+import org.quiltmc.enigma.api.translation.representation.entry.MethodEntry;
+import org.quiltmc.enigma.util.I18n;
+import org.quiltmc.enigma.util.Pair;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class MappingsIndex implements MappingsIndexer {
+	private final Collection<MappingsIndexer> indexers;
+	private final PackageIndex packageIndex;
+
+	private ProgressListener progress;
+	private int work;
+
+	public MappingsIndex(PackageIndex packageIndex) {
```

Comment:
I feel like this could be implemented as a `Map<Class<? extends MappingsIndexer>, MappingsIndexer>` for indexers collection, and here `MappingsIndexer... indexers` as a parameter, so you keep any arbitrary mapping indexers instead of having a field for each one of them... though I'm not sure if it'd useful at all...

Reply:
true, I just copied the way `JarIndex` does it. I'll change things up tomorrow!